### PR TITLE
Adding support for string table lookups in dropdown field labels

### PR DESCRIPTION
**Pull request by AnmAtAnm**
Adds `Blockly.utils.replaceMessageReferences()` method to handle string replacement without interpolation tokens.  Effectively uses the same old code, now moved into `tokenizeInterpolation_()`, which takes a parseInterpolationTokens option.

Replaces the direct JavaScript references (not pure JSON, and thus not portable).

Demonstrating this behavior in the `logic_boolean` dropdown.

---
This change is [Reviewable](https://reviewable.io/reviews/google/blockly/818)

### CODE REVIEW

File: core/utils.js [hunk 1]
```diff
@@ -392,11 +392,38 @@ Blockly.utils.commonWordSuffix = function(array, opt_shortest) {
...
+ * references (e.g., %{bky_my_msg} will be replaced with Blockly.Msg['MY_MSG']).
  * @param {string} message Text containing interpolation tokens.
  * @return {!Array.<string|number>} Array of strings and numbers.
  */
 Blockly.utils.tokenizeInterpolation = function(message) {
+  return Blockly.utils.tokenizeInterpolation_(message, true);
+}
+
+/**
+ * Parse a string with and replace string table references. For example,
```
Comment (rachel-fenichel) [inline core/utils.js]: I can't parse this comment.  Did you miss a word?

File: core/utils.js [hunk 2]
```diff
@@ -392,11 +392,38 @@ Blockly.utils.commonWordSuffix = function(array, opt_shortest) {
...
+}
+
+/**
+ * Internal implemention of the message reference and interpolation token
+ * parsing used by tokenizeInterpolation() and replaceMessageReferences().
+ * @param {string} message Text containing message references and interpolation
+ *     tokens.
+ * @param {boolean} parseInterpolationTokens Option to parse numeric
+ *     interpolation tokens (%1, %2, ...) when true.
+ * @return {!Array.<string|number>} Array of strings and numbers.
```
Comment (rachel-fenichel) [inline core/utils.js]: @private

**Comment by RoboErikG** [comment]
Aside from Rachel's comments LGTM.

**Comment by AnmAtAnm** [comment]
Review status: 0 of 4 files reviewed at latest revision, 2 unresolved discussions.

---

*[core/utils.js, line 405 at r1](https://reviewable.io:443/reviews/google/blockly/818#-KaEqdrKtNPpIVvGze47-r1-405:-KaEsSYXXdHPT6Fw33YB:bp0ozkv) ([raw file](https://github.com/google/blockly/blob/4bdaee0e3b2abfaf04e9b6ad3d1c0acdc38ed848/core/utils.js#L405)):*
<details><summary><i>Previously, rachel-fenichel (Rachel Fenichel) wrote…</i></summary><blockquote>

I can't parse this comment.  Did you miss a word?
</blockquote></details>

Corrected and rewrote. 

---

*[core/utils.js, line 424 at r1](https://reviewable.io:443/reviews/google/blockly/818#-KaEqdrKtNPpIVvGze47-r1-424:-KaEqf7a33TuOXDgNJHj:b-896fix) ([raw file](https://github.com/google/blockly/blob/4bdaee0e3b2abfaf04e9b6ad3d1c0acdc38ed848/core/utils.js#L424)):*
<details><summary><i>Previously, rachel-fenichel (Rachel Fenichel) wrote…</i></summary><blockquote>

@private
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly/818)*

**Comment by rachel-fenichel** [comment]
LGTM

**Comment by timdawborn** [comment]
This PR broke image support in dropdowns. See #848.

**Pull request merged**